File: android/app/src/main/java/dr/achim/sleep_timer/AlarmReceiver.java
package dr.achim.sleep_timer;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.embedding.engine.FlutterEngineCache;

import dr.achim.sleep_timer.Messages.*;

public class AlarmReceiver extends BroadcastReceiver {
    private static final String TAG = AlarmReceiver.class.toString();
    private Messages.FlutterTimerApi flutterTimerApi;

    @Override
    public void onReceive(Context context, Intent intent) {
        initializeFlutter();

        Log.e(TAG, "onReceive()");

        final String timerId = intent.getStringExtra(NotificationReceiver.KEY_TIMER_ID);
        TimerRequest request = new TimerRequest();
        request.setTimerId(timerId);
        flutterTimerApi.onAlarm(request, reply -> {});
    }

    private void initializeFlutter() {
        final FlutterEngine flutterEngine = FlutterEngineCache.getInstance().get(MainActivity.ENGINE_ID);
        if(flutterEngine != null) {
            flutterTimerApi = new Messages.FlutterTimerApi(flutterEngine.getDartExecutor().getBinaryMessenger());
        }
    }
}

File: android/app/src/main/kotlin/dr/achim/sleep_timer/AlarmReceiver.kt
package dr.achim.sleep_timer

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import dr.achim.sleep_timer.Messages.FlutterTimerApi
import dr.achim.sleep_timer.Messages.TimerRequest
import io.flutter.embedding.engine.FlutterEngineCache

class AlarmReceiver : BroadcastReceiver() {
    companion object {
        private val TAG = AlarmReceiver::class.java.toString()
    }

    private lateinit var flutterTimerApi: FlutterTimerApi

    override fun onReceive(context: Context, intent: Intent) {
        Log.d(TAG, "onReceive()")
        initializeFlutter()
        val request = TimerRequest().apply {
            timerId = intent.getStringExtra(NotificationReceiver.KEY_TIMER_ID)
        }
        flutterTimerApi.onAlarm(request) { }
    }

    private fun initializeFlutter() {
        FlutterEngineCache.getInstance()[MainActivity.ENGINE_ID]?.let {
            flutterTimerApi = FlutterTimerApi(it.dartExecutor.binaryMessenger)
        }
    }

}